src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/mainscreen/viewmodel/MainScreenViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.wycliffeassociates.otter.jvm.app.ui.mainscreen.viewmodel
+
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleStringProperty
+import org.wycliffeassociates.otter.common.data.model.ContentLabel
+import org.wycliffeassociates.otter.common.data.workbook.Chapter
+import org.wycliffeassociates.otter.common.data.workbook.Chunk
+import org.wycliffeassociates.otter.common.data.workbook.Workbook
+import org.wycliffeassociates.otter.jvm.app.ui.mainscreen.view.MainScreenView
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.view.CardGridFragment
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.view.TakeManagementFragment
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
+import tornadofx.*
+
+class MainScreenViewModel : ViewModel() {
+    private val workbookViewModel: WorkbookViewModel by inject()
+
+    val selectedProjectName = SimpleStringProperty()
+    val selectedProjectLanguage = SimpleStringProperty()
+
+    val selectedChapterTitle = SimpleStringProperty()
+    val selectedChapterBody = SimpleStringProperty()
+
+    val selectedChunkTitle = SimpleStringProperty()
+    val selectedChunkBody = SimpleStringProperty()
+
+    private val takesPageDocked = SimpleBooleanProperty(false)
+
+    init {
+        workbookViewModel.activeWorkbookProperty.onChange {
+            it?.let { wb -> projectSelected(wb) }
+        }
+
+        workbookViewModel.activeChapterProperty.onChange {
+            it?.let { chapter -> chapterSelected(chapter) }
+        }
+
+        workbookViewModel.activeChunkProperty.onChange {
+            it?.let { chunk -> chunkSelected(chunk) }
+                ?: takesPageDocked.set(false)
+        }
+    }
+
+    private fun projectSelected(selectedWorkbook: Workbook) {
+        setActiveProjectText(selectedWorkbook)
+
+        find<MainScreenView>().activeFragment.dock<CardGridFragment>()
+    }
+
+    private fun chapterSelected(chapter: Chapter) {
+        setActiveChapterText(chapter)
+    }
+
+    private fun chunkSelected(chunk: Chunk) {
+        setActiveChunkText(chunk)
+
+        if(takesPageDocked.value == false) {
+            find<MainScreenView>().activeFragment.dock<TakeManagementFragment>()
+        }
+        takesPageDocked.set(true)
+    }
+
+    private fun setActiveChunkText(chunk: Chunk) {
+        selectedChunkTitle.set(ContentLabel.of(chunk.contentType).value.toUpperCase())
```

Comment:
Internationalization / look up in Messages?

Reply:
I just talked to Thomas about those nav box cards, and we're not entirely sure how the nav box cards are going to work now that selecting a resource actually selects both a chunk and a resource. UNLESS the resource is about the chapter as a whole, which is a different case. It seems to me like this issue has to be deferred until we know what's really going to happen.

Reply:
What? 
In the mean time could you just put it in a messages lookup?

Reply:
That's fine, for now I just changed it to`   selectedChunkTitle.set(messages["verse"].toUpperCase())`